### PR TITLE
[Don't merge] Use registerWithSyncManager for places (#10128)

### PR DESCRIPTION
**Pull request by bendk**
This one switches over to using `registerWithSyncManager` for places syncing.  It depends on ~mozilla/application-services#4627~ https://github.com/mozilla/application-services/pull/4634.

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by bendk** [comment]
I made a new PR for the a-s issue and it's approved, we just want to get the consumer code ready for the change.  Can someone review this one so that we can move it forward?  Still don't merge, we need to make an a-s release first.

File: components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesHistoryStorage.kt
Comment (eliserichards) [inline components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesHistoryStorage.kt:236]: It would be great if we could add this in `PlacesHistoryStorageTest.kt`. Not sure how tricky that would be 🤔

Comment (eliserichards) [reply components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesHistoryStorage.kt:236]: In the same vein, are there any changes that should be made to `PlacesBookmarksStorageTest.kt` as well?

Comment (bendk) [reply components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesHistoryStorage.kt:236]: I would love love to test the sync manager integration more, but I can't think of how to do it exactly.  We could add a sync manager function that returns which engines are registered to and test that things seem to be linking up correctly.  Testing the actual sync process would be very tricky though.

Comment (bendk) [reply components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesHistoryStorage.kt:236]: Another issue with testing is that adding methods to sync manager is tricky before it's using UniFFI. I just updated the s[ync manager UniFFI PR](https://github.com/mozilla/application-services/pull/4645) with [some tests for `registerWithSyncManager`](https://github.com/mozilla/application-services/pull/4645/commits/d4ea9bc1d550d3d3035a04e7ed8b5bca03787f69)

This is the best I can think of.  It seems like that should test that places is getting registered with the sync manager correctly at least.  It would be great to test the actual syncing functionality, but that's a lot more work.

Do you think this one is in good enough shape that we can merge the a-s branch and make an a-s release with the new code?  Then we can test this better against the actual a-s code and make the CI pass.

**Review by eliserichards — CHANGES_REQUESTED** [review]
We'll also want to add a line in the changelog for this before it's merged

File: components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesHistoryStorage.kt
Comment (eliserichards) [reply components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesHistoryStorage.kt:236]: Yeah, let's go ahead and get the a-s branch merged 👍

**Comment by bendk** [comment]
We decided to make an A-S release that includes the change related to this and also the sync manager UniFFI work.  Let's use #11256 to track the rest of this work.

**Pull request closed without merging**
